testbed/datasenders/fluent.go

```diff
@@ -0,0 +1,160 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datasenders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/fluent/fluent-logger-golang/fluent"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+const (
+	fluentSplunkHECTestcseDatafileVar = "FLUENT_SPLUNKHEC_TESTCASE_DATA_FILE"
+	fluentPortVar                     = "FLUENT_SPLUNKHEC_FLUENT_PORT"
+)
+
+// FluentLogsForwarder forwards logs to fluent forwader
+type FluentLogsForwarder struct {
+	testbed.DataSenderBase
+	fluentLogger *fluent.Fluent
+	dataFile     *os.File
+}
+
+// Ensure FluentLogsForwarder implements LogDataSender.
+var _ testbed.LogDataSender = (*FluentLogsForwarder)(nil)
+
+func NewFluentLogsForwarder(t *testing.T, port int) *FluentLogsForwarder {
+	var err error
+	portOverride := os.Getenv(fluentPortVar)
+	if portOverride != "" {
+		port, err = strconv.Atoi(portOverride)
+		require.NoError(t, err)
+	}
+
+	f := &FluentLogsForwarder{DataSenderBase: testbed.DataSenderBase{Port: port}}
+
+	if dataFileName := os.Getenv(fluentSplunkHECTestcseDatafileVar); dataFileName != "" {
+		f.dataFile, err = os.OpenFile(dataFileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+		require.NoError(t, err)
+	} else {
+		logger, err := fluent.New(fluent.Config{FluentPort: port, Async: true})
+		require.NoError(t, err)
+		f.fluentLogger = logger
+	}
+	return f
+}
+
+func (f *FluentLogsForwarder) Start() error {
+	return nil
+}
+
+func (f *FluentLogsForwarder) Stop() error {
+	return f.fluentLogger.Close()
+}
+
+func (f *FluentLogsForwarder) ConsumeLogs(_ context.Context, logs pdata.Logs) error {
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		for j := 0; j < logs.ResourceLogs().At(i).InstrumentationLibraryLogs().Len(); j++ {
+			ills := logs.ResourceLogs().At(i).InstrumentationLibraryLogs().At(j)
```

Comment:
Are we ignoring `Resource` here?

Reply:
For the purposes of the test, we only care about the log records themselves.